server/src/main/java/org/apache/lucene/search/uhighlight/CustomFieldHighlighter.java

```diff
@@ -76,4 +82,84 @@
+            if (left.getScore() < right.getScore()) {
+                return -1;
+            } else if (left.getScore() > right.getScore()) {
+                return 1;
+            } else {
+                return left.getStartOffset() - right.getStartOffset();
+            }
+        });
+        Passage passage = new Passage(); // the current passage in-progress.  Will either get reset or added to queue.
+
+        do {
+            int start = off.startOffset();
+            if (start == -1) {
+                throw new IllegalArgumentException("field '" + field + "' was indexed without offsets, cannot highlight");
+            }
+            int end = off.endOffset();
+            if (start < contentLength && end > contentLength) {
+                continue;
+            }
+            // See if this term should be part of a new passage.
+            if (start >= passage.getEndOffset()) {
+                passage = maybeAddPassage(passageQueue, passageScorer, passage, contentLength);
+                // if we exceed limit, we are done
+                if (start >= contentLength) {
+                    break;
+                }
+                passage.setStartOffset(Math.max(this.breakIterator.preceding(start + 1), 0));
+                passage.setEndOffset(Math.min(this.breakIterator.following(start), contentLength));
+            }
+            // Add this term to the passage.
+            BytesRef term = off.getTerm();// a reference; safe to refer to
+            assert term != null;
+            passage.addMatch(start, end, term, off.freq());
+        } while (off.nextPosition());
+        maybeAddPassage(passageQueue, passageScorer, passage, contentLength);
+
+        Passage[] passages = passageQueue.toArray(new Passage[passageQueue.size()]);
+        // sort in ascending order
+        Arrays.sort(passages, Comparator.comparingInt(Passage::getStartOffset));
+        return passages;
+    }
+
+    // TODO: use FieldHighlighter::maybeAddPassage
+    // After removing CustomFieldHighlighter::highlightOffsetsEnums, remove this method as well.
+    private Passage maybeAddPassage(PriorityQueue<Passage> passageQueue, PassageScorer scorer, Passage passage, int contentLength) {
+        if (passage.getStartOffset() == -1) {
+            // empty passage, we can ignore it
+            return passage;
+        }
+        passage.setScore(scorer.score(passage, contentLength));
+        // new sentence: first add 'passage' to queue
+        if (passageQueue.size() == maxPassages && passage.getScore() < passageQueue.peek().getScore()) {
+            passage.reset(); // can't compete, just reset it
+        } else {
+            passageQueue.offer(passage);
+            if (passageQueue.size() > maxPassages) {
+                passage = passageQueue.poll();
+                passage.reset();
+            } else {
+                passage = new Passage();
+            }
+        }
+        return passage;
+    }
```

Comment:
This just overrides LUCENE-9093 and preserves the previous behaviour, correct? +1 if so

Reply:
exactly, it just preserves the previous behaviour.